### DISCUSSION TITLE
Problema de acesso Content-type application-x-www-form-urlencoded

### DISCUSSION THREAD

**Discussion by cezarmolinar** [viniciussanchez/RESTRequest4Delphi#170] [General]
Estou tentando acessar uma API onde o content-type não é json

Com TRestClient/TRestRequest consigo acesso com este código:
  RESTClient.BaseURL := 'https://sandbox-api.agendaedu.dev/';
  RESTRequest.Resource := 'oauth/v2/token';
  RESTRequest.Method := TRESTRequestMethod.rmPOST;
  RESTRequest.Params.AddItem('Content-Type', 'application-x-www-form-urlencoded', pkREQUESTBODY, []);
  RESTRequest.Params.AddItem('grant_type', 'client_credentials', pkREQUESTBODY, []);
  RESTRequest.Params.AddItem('client_id', 'xxxxxxxxxxx', pkREQUESTBODY, []);
  RESTRequest.Params.AddItem('client_secret', 'yyyyyyyy', pkREQUESTBODY, []);
  RESTRequest.Execute;

Tentei com o RESTRequest4D com esse código:
  lResponse := TRequest.New
    .BaseURL('https://sandbox-api.agendaedu.dev/')
    .Resource('oauth/v2/token')
    .ContentType('application-x-www-form-urlencoded')
    .AddParam('Content-Type', 'application-x-www-form-urlencoded', pkREQUESTBODY)
    .AddParam('grant_type', 'client_credentials', pkREQUESTBODY)
    .AddParam('client_id', 'xxxxxxxxxxx', pkREQUESTBODY)
    .AddParam('client_secret', 'yyyyyyyy', pkREQUESTBODY)
    .Post;

Mas sempre me retorno 401 (não autorizado)

**Comment by viniciussanchez** [comment]
Tente assim:

lResponse := TRequest.New
.BaseURL('https://sandbox-api.agendaedu.dev/')
.Resource('oauth/v2/token')
.AddParam('Content-Type', 'application-x-www-form-urlencoded', pkREQUESTBODY)
.AddParam('grant_type', 'client_credentials', pkREQUESTBODY)
.AddParam('client_id', 'xxxxxxxxxxx', pkREQUESTBODY)
.AddParam('client_secret', 'yyyyyyyy', pkREQUESTBODY)
.Post;

ou se descobrir o motivo, envie um pull request, ou issue para a gente ajustar
